<center><font color=#555555><font size=6> MAF1731 - Microstructure and Trading Systems </font> <br> <br>
    
<center><font color=#555555><font size=5> Ingeniería Financiera </font> <br> <br> <br> <br>

<center><font color=#555555><font size=6> I.F. Cuauhtémoc Corrales Camacho (CCC) </font> <br> <br>
<center><font color=#555555><font size=6> I.F. Esteban Marquez Delgado (EMD) </font> <br> <br>
<center><font color=#555555><font size=6> I.F. Marco Antonio Ochoa Cárdenas (MAOC) </font> <br> <br>
   
<center> <font color= #555555> <font size = 4> Octubre 2022 </a> | <font color= #555555> <font size = 4> Repository: <a href='https://github.com/CuauhtemocCC/MyST_LAB_3_E3'>Link</a></font>

<hr style="border:0.02in solid gray"> </hr>

<center> <font color= #555555> <font size = 6> Lab 3 </font> <br> <br> <font color= #555555> <font size = 5> Comportamiento Financiero  </font>

<hr style="border:0.02in solid gray"> </hr>

# <font color= #6B6B6B> <font size = 6> 1. Introducción </font>

<hr style="border:0.02in solid gray"> </hr>

Dia a día en el mundo de las inversiones, existen mil y un ofertas para colocar nuestros recursos económicos, sin embargo, como podemos estar seguros de que la persona a cargo es un guru de las finanzas o tal vez solamente un tipo con un curso de trading por internet, tampoco es debamos desconfiar de cualquier persona o institución, hay empresas e individuos que han dedicado una enorme cantidad de tiempo a formarse una reputación, pero al final, al final de todo, los que llevan a cabo la toma de decisiones en este mundo son humanos y como humanos inherentemente tienen sentimientos y prejucios parecidos a los de los demás, tampoco es que cualquier persona pueda dedicarse al trading claro está, pero si podemos evaluar y traer a la consciencia los sesgos de conducata más comunes que tienden a cometer los traders, para posteriormente generar estrategias que ayuden a contrarestarlos, pero al mismo tiempo por parte de los clientes poder evalaur a los gestores.

Dentro de las inversiones podemos tener muchas maneras en las que podemos estar invirtiendo con aversiones al riesgo y en búsqueda de distintos rendimientos dependiendo de ellos, una vez que se tiene pensado invertir en instrumentos bursátiles que pueden tener un porcentaje de riesgo o con una aversión media alta al riesgo podemos optar en el trading como una buena opción para invertir, dentro del trading podemos destacar algunas definiciones que son importantes para conocer un poco más acerca del tema. En este laboratorio se tienen algunas definiciones como el draw up y draw down (se definirán más adelante en el proyecto). Uno de los puntos más importantes es la inversión racional la cual se define como aquella inversión que busca maximizar el beneficio entre alternativas de igual nivel de riesgo es decir elegir la inversión que más rendimiento nos pueda dar teniendo el mínimo riesgo. 
Otra de las definiciones que tenemos que conocer en este laboratorio es el sesgo de inversión el cual se define como la tendencia a sobreestimar los conocimientos y juicios subjetivos y considerarlos certeros. A la hora de tomar decisiones y realizar pronósticos, se sobrevaloran los conocimientos y la experiencia personal sin tener en cuenta la diferencia entre lo que se sabe realmente y lo que se cree saber.


# <font color= #6B6B6B> 2. Objetivos </font>

<hr style="border:0.02in solid gray"> </hr>

Analizar datos sobre posiciones de trading con el fin de detectar sesgos de comportamientos en dichas operaciones.

# <font color= #6B6B6B> 3. Descripción de Datos </font>

<hr style="border:0.02in solid gray"> </hr>

In [1]:
# Importacion de Paqueterias

import data as dt
import visualizations as vs
import pandas as pd
import functions as fn

[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed


## <font color= #6B6B6B> 3.1 Importación: Tablas de posiciones </font>

Para dar inicio con este proyecto se requerió de nuestras tablas de posiciones obtenidos en la elaboracion del Laboratorio 2, donde durante un periodo de tiempo determinado se desarrollo un metodo de trading para cada integrante. 


In [2]:
# Segmento base de posiciones 
dt.datos_c.head(3)

Fecha/Hora  Posición Símbolo  Tipo  Volumen    Precio     S / L  \
0  2022.09.15 21:17:28  58586674  EURUSD   buy     0.01    0.9995    0.9992   
1  2022.09.15 21:21:39  58586767  EURUSD   buy     0.10    0.9998    0.9995   
2  2022.09.16 21:52:23  58630858  USDJPY  sell     0.10  142.9020  142.9400   

       T / P         Fecha/Hora.1   Precio.1  ...  profit_acum  Ops Totales  \
0    0.99955  2022.09.15 21:18:35    0.99955  ...         0.05          1.0   
1    0.99980  2022.09.15 21:23:43    0.99950  ...        -2.95          1.0   
2  142.82000  2022.09.16 21:54:30  142.94000  ...        -5.61          1.0   

   Ops Ganadoras  Ops Perdedoras Ops Ganadoras_C Ops Perdedoras_C  \
0            1.0             0.0             1.0              0.0   
1            0.0             1.0             0.0              1.0   
2            0.0             1.0             0.0              0.0   

  Ops Ganadoras_V  Ops Perdedoras_V         day    Capital  
0             0.0               0.0  2022-09-15  100000.05  
1             0.0               0.0  2022-09-15   99997.05  
2             0.0               1.0  2022-09-16   99994.39  

[3 rows x 29 columns]

Con la tabla de posiciones, se siguieron instrucciones de implementar dos cambios extras, el primero fue adaptar las columnas de precio y precio.1 como *opentime* y *closetime*, para volcer dichas importantes variables indetificables y en el formato adecuado para el resto del proceso. 

In [3]:
# Cambio 2
dt.datos2_c.head(3)

Fecha/Hora  Posición Símbolo  Tipo  Volumen    Precio     S / L  \
0  2022.09.15 21:17:28  58586674  EURUSD   buy     0.01    0.9995    0.9992   
1  2022.09.15 21:21:39  58586767  EURUSD   buy     0.10    0.9998    0.9995   
2  2022.09.16 21:52:23  58630858  USDJPY  sell     0.10  142.9020  142.9400   

       T / P         Fecha/Hora.1   Precio.1  ...  profit_acum  Ops Totales  \
0    0.99955  2022.09.15 21:18:35    0.99955  ...         0.05          1.0   
1    0.99980  2022.09.15 21:23:43    0.99950  ...        -2.95          1.0   
2  142.82000  2022.09.16 21:54:30  142.94000  ...        -5.61          1.0   

   Ops Ganadoras  Ops Perdedoras Ops Ganadoras_C Ops Perdedoras_C  \
0            1.0             0.0             1.0              0.0   
1            0.0             1.0             0.0              1.0   
2            0.0             1.0             0.0              0.0   

  Ops Ganadoras_V  Ops Perdedoras_V         day    Capital  
0             0.0               0.0  2022-09-15  100000.05  
1             0.0               0.0  2022-09-15   99997.05  
2             0.0               1.0  2022-09-16   99994.39  

[3 rows x 29 columns]

El siguiente paso fue establecer metricas atribuidas al beneficio acumulado en le capital tras cada posicion asi como tambien el calculo de los pips que se ganaron o pedieron tras cada operacion y establecer un cuenta que contara su acumulado.

In [4]:
# Cambio 3
dt.datos3_c.head(3)

Fecha/Hora  Posición Símbolo  Tipo  Volumen    Precio     S / L  \
0  2022.09.15 21:17:28  58586674  EURUSD   buy     0.01    0.9995    0.9992   
1  2022.09.15 21:21:39  58586767  EURUSD   buy     0.10    0.9998    0.9995   
2  2022.09.16 21:52:23  58630858  USDJPY  sell     0.10  142.9020  142.9400   

       T / P         Fecha/Hora.1   Precio.1  ...  profit_acum  Ops Totales  \
0    0.99955  2022.09.15 21:18:35    0.99955  ...         0.05          1.0   
1    0.99980  2022.09.15 21:23:43    0.99950  ...        -2.95          1.0   
2  142.82000  2022.09.16 21:54:30  142.94000  ...        -5.61          1.0   

   Ops Ganadoras  Ops Perdedoras Ops Ganadoras_C Ops Perdedoras_C  \
0            1.0             0.0             1.0              0.0   
1            0.0             1.0             0.0              1.0   
2            0.0             1.0             0.0              0.0   

  Ops Ganadoras_V  Ops Perdedoras_V         day    Capital  
0             0.0               0.0  2022-09-15  100000.05  
1             0.0               0.0  2022-09-15   99997.05  
2             0.0               1.0  2022-09-16   99994.39  

[3 rows x 29 columns]

Ademas de estos procesos, se establecieron algunas variables de soporte que seran de ayuda para los siguientes procesos, la mayoria de estas se identifican como vectores de conteo de 1s.
Al final de todos los preparativos, esta columnas quedaron en nuestra tabla base.

In [5]:
# Columnas tabla base
dt.datos3_c.columns

Index(['Fecha/Hora', 'Posición', 'Símbolo', 'Tipo', 'Volumen', 'Precio',
       'S / L', 'T / P', 'Fecha/Hora.1', 'Precio.1', 'Comisión', 'Swap',
       'Beneficio', 'Lote', 'opentime', 'closetime', 'time(seconds)', 'pips',
       'pips_acum', 'profit_acum', 'Ops Totales', 'Ops Ganadoras',
       'Ops Perdedoras', 'Ops Ganadoras_C', 'Ops Perdedoras_C',
       'Ops Ganadoras_V', 'Ops Perdedoras_V', 'day', 'Capital'],
      dtype='object')

# <font color= #6B6B6B> 4. Procesos  </font>

<hr style="border:0.02in solid gray"> </hr>

A continuación se describiran los procesos realizados para gestionar el analisis de cada una de las posiciones con el fin de obtener metricas conclusivas de medicion, asi como definir el tipo de comportamiento que cada integrante desarrollo al momento de realizar su trading correspondiente.

## <font color= #6B6B6B> 4.1 Tabla Operaciones  </font>

A partir de nuestra base inicial establecida, el primer analisis sera uno del tipo descriptivo, donde se contaran las operaciones ganadoras y perdedoras en trading de cada uno de los integrantes, asi como el rendimiento en pips y pondereciones entre los tres conceptos, tales como serian la efectividad total de las operaciones, las tasas de proporcion de operaciones ganadoras y perdedoras sobre el total, y la ganancia media y mediana de las inversiones.

In [6]:
# Tabla decriptiva de operaciones y beneficios, CCC
dt.df_1_tabla_c

medidas   valor                            descripcion
0        Ops totales  28.000                    Operaciones totales
1          Ganadoras  14.000                  Operaciones ganadoras
2        Ganadoras_c   5.000        Operaciones ganadoras de compra
3        Ganadoras_v   9.000        Operaciones perdedoras de venta
4         Perdedoras  14.000                 Operaciones perdedoras
5       Perdedoras_c   6.000       Operaciones perdedoras de compra
6       Perdedoras_v   8.000        Operaciones perdedoras de venta
7   Mediana (Profit)   0.025       Mediana de profit de operaciones
8     Mediana (Pips)   0.025         Mediana de pips de operaciones
9      r_efectividad   0.500  Ganadoras Totales/Operaciones Totales
10      r_proporcion   1.000   Ganadoras Totales/Perdedoras Totales
11   r_efectividad_c   0.179  Ganadoras Compras/Operaciones Totales
12   r_efectividad_v   0.321  Ganadoras Ventas/ Operaciones Totales

In [7]:
# Tabla decriptiva de operaciones y beneficios, EMD
dt.df_1_tabla_e

medidas   valor                            descripcion
0        Ops totales  80.000                    Operaciones totales
1          Ganadoras  63.000                  Operaciones ganadoras
2        Ganadoras_c  60.000        Operaciones ganadoras de compra
3        Ganadoras_v   3.000        Operaciones perdedoras de venta
4         Perdedoras  17.000                 Operaciones perdedoras
5       Perdedoras_c  17.000       Operaciones perdedoras de compra
6       Perdedoras_v   0.000        Operaciones perdedoras de venta
7   Mediana (Profit)   0.310       Mediana de profit de operaciones
8     Mediana (Pips)   0.320         Mediana de pips de operaciones
9      r_efectividad   0.787  Ganadoras Totales/Operaciones Totales
10      r_proporcion   1.000   Ganadoras Totales/Perdedoras Totales
11   r_efectividad_c   0.750  Ganadoras Compras/Operaciones Totales
12   r_efectividad_v   0.037  Ganadoras Ventas/ Operaciones Totales

In [8]:
# Tabla decriptiva de operaciones y beneficios, MAOC
dt.df_1_tabla_m

medidas   valor                            descripcion
0        Ops totales  33.000                    Operaciones totales
1          Ganadoras   9.000                  Operaciones ganadoras
2        Ganadoras_c   7.000        Operaciones ganadoras de compra
3        Ganadoras_v   2.000        Operaciones perdedoras de venta
4         Perdedoras  24.000                 Operaciones perdedoras
5       Perdedoras_c  20.000       Operaciones perdedoras de compra
6       Perdedoras_v   4.000        Operaciones perdedoras de venta
7   Mediana (Profit) -59.840       Mediana de profit de operaciones
8     Mediana (Pips) -80.200         Mediana de pips de operaciones
9      r_efectividad   0.273  Ganadoras Totales/Operaciones Totales
10      r_proporcion   1.000   Ganadoras Totales/Perdedoras Totales
11   r_efectividad_c   0.212  Ganadoras Compras/Operaciones Totales
12   r_efectividad_v   0.061  Ganadoras Ventas/ Operaciones Totales

De esta apartado se puede clasificar que los metodos de inversion utilizados, por ejemplo, Esteban logro una efectividad del 78.7%, donde se aposto por operaciones de compra; Cuauhtemoc del 50%, quien aposto mas por operaciones de venta; y Marco del 27.3%, ademas de que fue el unico que tuvo una mediana de pips en valores negativos.

## <font color= #6B6B6B> 4.2 Tabla Ranking de Efectividad  </font>

 En esta tabla se observa el margen de efectividad, a manera de porcentaje, sobre los instrumentos que cada integrante utilizo. Es decir, de todas las operaciones generadas con un instrumento, que proporcion fueron ganadoras.

In [9]:
# Tabla ranking, CCC
dt.df_2_tabla_c

rank   rank %
Símbolo               
GBPJPY   1.00  100.00%
USDJPY   0.80   80.00%
EURGBP   0.67   67.00%
EURUSD   0.38   38.00%
AUDUSD   0.33   33.00%
EURMXN   0.00    0.00%
GBPUSD   0.00    0.00%
USDCAD   0.00    0.00%

In [18]:
# Tabla ranking, EMD
dt.df_2_tabla_e

rank  rank %
Símbolo              
EURUSD   0.91  91.00%
USDMXN   0.65  65.00%
GBPUSD   0.00   0.00%

In [10]:
# Tabla ranking, MAOC
dt.df_2_tabla_m

rank   rank %
Símbolo                  
TSLA.O      1.00  100.00%
#WHEAT_Z22  0.50   50.00%
GOLD        0.40   40.00%
#USSPX500   0.38   38.00%
#USNDAQ100  0.33   33.00%
WTI         0.20   20.00%
EURUSD      0.00    0.00%
GBPUSD      0.00    0.00%
SILVER      0.00    0.00%
SNAP.N      0.00    0.00%
USDCAD      0.00    0.00%
USDMXN      0.00    0.00%

Algo a destacar en esta parte, es que los tres individuos utilizaron instrumentos que no les dejo ninguna rentabilidad, lo cual quiere decir que su estrategia de inversion fue mala; por otro lado, tanto Marco como Cuauhtemoc tienen una lista de instrumentos mas diversa, lo cual deberia de disminuir el riesgo total de perdida, e incluso un instrumento con el 100% de efectividad, por otra parte, Esteban solo trabajo con tres instrumentos y esto pudo haber influido en su alto margen de efectividad, ya que al manejar pocos instrumentos se puede estudiar mejor su comportamiento.

## <font color= #6B6B6B> 4.3 Tabla evolucion del capital  </font>

En esta tabla se describen los cambios que sufrio el capital inicial de 100,000 dolares a lo largo de los dias de inversion. Esta tabla sirve para ver el comportamiento de la cuenta base de cada integrante a traves de sus inversiones correspodientes. Para este resultado se agrupo de manera diaria el beneficio neto obtenido y se agrego a la cuenta del capital. Los nombres de las columnas representan:
+ **timestamp:** dia correspodiente de inversiones

+ **profit_d:** beneficio neto obtenido de las inversiones del dia

+ **profit_acum_d:** estado del capital base al final del dia

In [12]:
print(dt.evo_c_c)

    timestamp  profit_d  profit_acum_d  rends_log
0  2022-09-15     -2.95       99997.05  -0.000030
1  2022-09-16     -2.52       99994.53  -0.000025
2  2022-09-17      0.00       99994.53  -0.000000
3  2022-09-18      0.00       99994.53  -0.000000
4  2022-09-19     32.14      100026.67   0.000321
5  2022-09-20   -180.40       99846.27  -0.001805
6  2022-09-21      0.49       99846.76   0.000005
7  2022-09-22    -42.50       99804.26  -0.000426
8  2022-09-23      1.30       99805.56   0.000013


In [13]:
# Tabla evolucion de Capital, CCC
#dt.evo_c_c.drop(["DU","DD","rends_log"],axis=1)
dt.evo_c_c.drop(["rends_log"],axis=1)

timestamp  profit_d  profit_acum_d
0  2022-09-15     -2.95       99997.05
1  2022-09-16     -2.52       99994.53
2  2022-09-17      0.00       99994.53
3  2022-09-18      0.00       99994.53
4  2022-09-19     32.14      100026.67
5  2022-09-20   -180.40       99846.27
6  2022-09-21      0.49       99846.76
7  2022-09-22    -42.50       99804.26
8  2022-09-23      1.30       99805.56

In [14]:
# Tabla evolucion de Capital, EMD
#dt.evo_c_c.drop(["DU","DD","rends_log"],axis=1)
dt.evo_c_e.drop(["rends_log"],axis=1)

timestamp  profit_d  profit_acum_d
0  2022-09-16     57.63      100057.63
1  2022-09-17      0.00      100057.63
2  2022-09-18      0.00      100057.63
3  2022-09-19    334.56      100392.19
4  2022-09-20      0.00      100392.19
5  2022-09-21    -88.86      100303.33
6  2022-09-22      0.00      100303.33
7  2022-09-23     13.82      100317.15

In [15]:
# Tabla evolucion de Capital, MAOC
#dt.evo_c_c.drop(["DU","DD","rends_log"],axis=1)
dt.evo_c_m.drop(["rends_log"],axis=1)

timestamp  profit_d  profit_acum_d
0   2022-09-16    -57.60       99942.40
1   2022-09-17      0.00       99942.40
2   2022-09-18      0.00       99942.40
3   2022-09-19     22.47       99964.87
4   2022-09-20   -883.73       99081.14
5   2022-09-21   -568.75       98512.39
6   2022-09-22      2.16       98514.55
7   2022-09-23   1570.90      100085.45
8   2022-09-24      0.00      100085.45
9   2022-09-25      0.00      100085.45
10  2022-09-26  -9298.90       90786.55

En este ejericicio, se observa que al final de periodo de inversion, solo uno de los individuos termino con beneficios positivos sobre el capital inicial. Ademas tambien se observa de Marco realizo inversiones mas agresivas debido que el nivel de sus numeros es superior al del resto de los intergrantes. 

## <font color= #6B6B6B> 4.4 Tabla de Metricas de Atribucion al Desempeño  </font>

En esta tabla se representan metricas que califican el desempeño de cada reporte de posiciones. Con el proposito de calificar el trabajo de cada integrante realizado en el laboratorio 2. Las metricas aplicadas fueron: 

+ **Sharpe:** Una metrica que compara el desempeño de una inversion contra el valor de la tasa libre de riesgo, donde si el valor es positivo es que esta sobre la tasa y si es negativo esta por debajo. Para calificar nuestras posiciones se utilizo dos metodologias de calculo diferente, el *original* donde se utilizan el rendimiento logaritmico promedio de las inversiones, su desviacion estandar y la tasa libre de riesgo; y el *ajustado* donde se agrega a los calculos la media de los rendimientos logaritmicos de algun indice, en este caso el S&P500.

+ **Draw Down:** Representa la “racha” de en la que se presentó una perdida drástica después de haber llegado al máximo punto de evolución de capital en un periodo de inversión. 

+ **Dram Up:** Señala el lapso de tiempo donde se tuvo un alza continua después de haber tenido la caída Máxima del periodo que se analiza.

Como nota, se debe mencionar que las metricas de Sharpe se encuentran en porcentaje * 100, y los Draws en capital (usd).

In [16]:
# Tabla MAD, CCC
dt.metricas_c

Metrica          Tipo de Dato                Valor  \
0     sharpe_original              Cantidad             -80.3602   
1  sharpe_actualizado              Cantidad             -80.3304   
2       drawdown_capi         Fecha Inicial  2022-09-22 22:47:53   
3       drawdown_capi           Fecha Final  2022-09-22 22:50:10   
4       drawdown_capi  DrawDown $ (capital)              99804.3   
5         drawup_capi         Fecha Inicial  2022-09-16 22:03:38   
6         drawup_capi           Fecha Final  2022-09-19 02:37:19   
7         drawup_capi    DrawUp $ (capital)               100035   

                             Descripcion  
0          Sharpe Ratio Fórmula Original  
1          Sharpe Ratio Fórmula Ajustada  
2  Fecha inicial del DrawDown de Capital  
3    Fecha final del DrawDown de Capital  
4     Máxima pérdida flotante registrada  
5    Fecha inicial del DrawUp de Capital  
6      Fecha final del DrawUp de Capital  
7    Máxima ganancia flotante registrada

In [17]:
# Tabla MAD, EMD
dt.metricas_e

Metrica          Tipo de Dato                Valor  \
0     sharpe_original              Cantidad             -39.5263   
1  sharpe_actualizado              Cantidad             -39.5114   
2       drawdown_capi         Fecha Inicial  2022-09-14 16:24:15   
3       drawdown_capi           Fecha Final  2022-09-16 23:54:14   
4       drawdown_capi  DrawDown $ (capital)              99995.6   
5         drawup_capi         Fecha Inicial  2022-09-19 16:56:35   
6         drawup_capi           Fecha Final  2022-09-19 17:13:19   
7         drawup_capi    DrawUp $ (capital)               100392   

                             Descripcion  
0          Sharpe Ratio Fórmula Original  
1          Sharpe Ratio Fórmula Ajustada  
2  Fecha inicial del DrawDown de Capital  
3    Fecha final del DrawDown de Capital  
4     Máxima pérdida flotante registrada  
5    Fecha inicial del DrawUp de Capital  
6      Fecha final del DrawUp de Capital  
7    Máxima ganancia flotante registrada

In [18]:
# Tabla MAD, MAOC
dt.metricas_m

Metrica          Tipo de Dato                Valor  \
0     sharpe_original              Cantidad             -1.95717   
1  sharpe_actualizado              Cantidad             -1.95655   
2       drawdown_capi         Fecha Inicial  2022-09-26 18:27:04   
3       drawdown_capi           Fecha Final  2022-09-26 18:45:12   
4       drawdown_capi  DrawDown $ (capital)              90786.6   
5         drawup_capi         Fecha Inicial  2022-09-19 17:43:12   
6         drawup_capi           Fecha Final  2022-09-19 22:56:49   
7         drawup_capi    DrawUp $ (capital)              99964.9   

                             Descripcion  
0          Sharpe Ratio Fórmula Original  
1          Sharpe Ratio Fórmula Ajustada  
2  Fecha inicial del DrawDown de Capital  
3    Fecha final del DrawDown de Capital  
4     Máxima pérdida flotante registrada  
5    Fecha inicial del DrawUp de Capital  
6      Fecha final del DrawUp de Capital  
7    Máxima ganancia flotante registrada

Como respuesta a esto, se tiene que los rendimientos generados por los tres intregrantes no superan la tasa libre de riesgo, lo cual se observa al notar que los valores de las metricas de Sharpe de los 3 reportes estan por debajo de 0. 

## <font color= #6B6B6B> 4.5 Tabla Comportamiento Financiero y sesgos de Inversion </font>

La tabla de desempeño fue creada como un dataframe que almacena y expone variables creadas que califican el comportamiento de cada inversion. Dichas variables son el rendimiento mensual promedio, el rendimiento mensual acumulado final y el indice de sharpe; cada variable fue aplicada a los resultados de cada estrategia de inversion.

In [19]:
# Tabla Ocurrencias y Comportamiento de inversion
fn.f_be_de(dt.datos4)

{'Ocurrencias': {'Cantidad': 14,
  'Ocurrencia 1': {'timestamp': Timestamp('2022-09-15 21:18:35'),
   'Operaciones': {'Ganadoras': {'Instrumento': 0.0,
     'Volumen': 0.0,
     'Sentido': 0.0,
     'Profit_ganadora': 0.0},
    'Perdedoras': {'Instrumento': 0.0,
     'Volumen': 0.0,
     'Sentido': 0.0,
     'Profit_ganadora': 0.0}},
   'ratio_cp_profit_acm': nan,
   'ratio_cg_profit_acm': nan,
   'ratio_cp_cg': nan},
  'Ocurrencia 2': {'timestamp': Timestamp('2022-09-16 21:56:01'),
   'Operaciones': {'Ganadoras': {'Instrumento': 0.0,
     'Volumen': 0.0,
     'Sentido': 0.0,
     'Profit_ganadora': 0.0},
    'Perdedoras': {'Instrumento': 0.0,
     'Volumen': 0.0,
     'Sentido': 0.0,
     'Profit_ganadora': 0.0}},
   'ratio_cp_profit_acm': nan,
   'ratio_cg_profit_acm': nan,
   'ratio_cp_cg': nan},
  'Ocurrencia 3': {'timestamp': Timestamp('2022-09-20 01:13:37'),
   'Operaciones': {'Ganadoras': {'Instrumento': 0.0,
     'Volumen': 0.0,
     'Sentido': 0.0,
     'Profit_ganadora': 0.0}

# <font color= #6B6B6B> <font size = 6> 5. Conclusiones </font>

<hr style="border:0.02in solid gray"> </hr>

Como inicio de este apartado, se recuperaran los resultados obtenidos en cada tabla descriptiva, asi como se hara uso de algunos graficos con el fin de visualizar dichos resultados de una manera mas clara.

Para empezar, que claro que para poder hacer un analisis mas representativo seria conveniente tener un historico con mayor cantidad de operaciones, pero a pesar de esto se pudo exprimir suficiente informacion que apoya a conocer si las estrategias aplicadas en el trading son fructiferas o no estan logrando un beneficio positivo para nuestra inversion. Es de vital importancia en este nivel de inversion que un cliente y un asesor esten completamente de acuerdo con las inversiones o los riesgos que en este caso el trader puede efectuar, como vimos en algunos de los Excel de los compañeros fue complicado en 2 semanas generar rendimientos positivos y como era de esperar nuestro radio de sharpe fue negativo en menor o mayor proporcion, un ejemplo de esto fue uno de los trades que se tuvieron tuvo un sharpe de -1.9 aproximadamente, esto quiere decir que es casi 200 veces menos efectico que la tasa libre de riesgos que te pueda ofrecer instrumentos gubernamentales de alrededor de 8% o incluso deuda gubernamental que puede estar sobre TIIe + 1.5%, es aqui donde un inversor tiene que estar consciente de este tema ya que no es nada facil superar la tasas que podria darte una inversion aceptando el minimo de riesgo y que la ganancia en porcentaje puede ser abismal ya que se pueden generar rendimientos de mas del 40% anual este rendimiento tambien puede ser negativo y nuestro cliente debe tener eso en cuenta si es que esta dispuesto a arriesgar una inversion de esa manera, con los diferentes metodos de trading que pudimos tener en el laboratorio 2 junto con este laboratorio 3 se tuvieron muchos aprendizajes, nos pudimos dar cuenta de las diferentes tecnicas que podemos tener a la hora de hacer trading discrecional con analisis tecnico y el como analizar estos trades con la estadistica descriptiva que nos dio una mejor idea de como nos fue con las diferentes tecnicas que vimos en el laboratorio pasados y algunas otras tecnicas como las visualizaciones que nos apoyan en mostrar de mejor manera como es que nos fue durante estos dias que se estuvo haciendo trading.

## 5.1 Ponderacion operaciones ganadoras

En este grafico se pueden apreciar el margen de operaciones ganadoras que realizo cada individuo y como dichas operaciones se distribuyen en los distintos instrumentos de inversion de que cada quien decidio por utilizar. A primera instancia, la mayor parte de los instrumentos utlizados pertencen al area de Forex y divisas, esto debido a que la pagina de la pagina web FX Pro da de facto dichos instrumentos. 

In [20]:
# Ponderacion operaciones ganadoras, CCC
vs.plot_pie_graph(dt.df_2_tabla_c)

In [21]:
# Ponderacion operaciones ganadoras, EMD
vs.plot_pie_graph(dt.df_2_tabla_e)

In [22]:
# Ponderacion operaciones ganadoras, MAOC
vs.plot_pie_graph(dt.df_2_tabla_m)

En estos graficos se presenta a primera instancia la diversidad de operaciones entre los integrantes del equipo, y como la mayor podenracion de aquellos con mayor diversidad se encuentra en el 30%, donde dicho intrumento obtuvo el 100% de efectividad en sus inversiones. Por otro lado, el miembro restante tiene un reparto mas igualitario entre sus instrumentos efectivos. 

## 5.2 Visualizacion del Draw Up y Draw Down

En este parte, se tienen dos conceptos, el Draw Up y Draw Down, los cuales hacen referencia a los puntos maximos y minimos por los cuales paso el estado del capital a lo largo del periodo en el que se implementaron las operaciones. Dichos conceptos ayudan a entender la dispersion del capital en un rango de tiempo, asi como las posibles ganancias o perdidas potenciales experimentadas. Cada metodologia de inversion trajo consigo distintos puntos maximos y minimos de capital, incluso en diferncia con el capital inicial. 

Como nota, se ha de decir que el intervalo en el cual se instauraron los puntos o rangos, segun su caso, de DrawDown y DrawUp es **diario**. 

In [27]:
import main as mn

In [28]:
# Grafica evolucion de capital, CCC
vs.plot_DDC_DUC(dt.evo_c_c)

In [24]:
# Grafica evolucion de capital, EMD
vs.plot_DDC_DUC(dt.evo_c_e)

In [25]:
# Grafica evolucion de capital, MAOC
vs.plot_DDC_DUC(dt.evo_c_m)

En este segundo grupo de graficos donde se observan el comportamiento del capital hay dos puntos interesantes a destacar. 

+ El primer punto es que mientras uno de los integrantes tiene sus dos puntos de Draw Up y Drown Down estaticos, los otros poseen rangos de capital, lo cual puede interpertarse como ausencia de inversion en ese periodo de tiempo o como una gran coincidencia en los valores del capital. 

+ El segundo punto es que los tres integrantes tienen sus puntos en distintos niveles a partir del capital inicial. Cuauhtemoc posee su DU por encima del capital inicial y su DD por debajo del mismo,  Esteban presenta tanto su DU como DD por encima del capital inicial, y por el contrario, Marco presenta su DU y DD por debajo del capital inicial. Este es un punto interesante, ya que se aprecia la tendencia del capital de manera intriseca. 

# <font color= #6B6B6B> <font size = 6> 6. References </font>

<hr style="border:0.02in solid gray"> </hr>

Proyecto Laboratorio 3. https://github.com/CuauhtemocCC/MyST_LAB_3_E3. (2022).

<br>